pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,617 @@
+			expectedLength:    1,
+			expectedTimestamp: &podLocalTimestamps{CreationTimestamp: refTime, DeletionTimestamp: &refTime},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClock := clock.NewFakeClock(refTime)
+			podStore := NewPodStoreWithClock(podInformer, fakeClock)
+			err := podStore.pods.Add(pod1)
+			assert.NoError(t, err)
+			podStore.timestampMap = map[types.UID]*podLocalTimestamps{"pod1": {CreationTimestamp: refTime}}
+
+			expectedDeleteTime := refTime.Add(tt.timeWait)
+			podStore.onPodDelete(tt.obj)
+			assert.Equal(t, tt.expectedTimestamp, podStore.timestampMap["pod1"])
+			if tt.expectedLength != 0 {
+				assert.NotNil(t, podStore.timestampMap["pod1"].DeletionTimestamp)
+				fakeClock.SetTime(expectedDeleteTime.Add(-10 * time.Millisecond))
+				assert.Eventuallyf(t, func() bool {
+					return podStore.podsToDelete.Len() == 0
+				}, 1*time.Second, 10*time.Millisecond, "Pod should not be added to PodsToDelete")
+				fakeClock.SetTime(expectedDeleteTime.Add(10 * time.Millisecond))
+				assert.Eventuallyf(t, func() bool {
+					return podStore.podsToDelete.Len() == tt.expectedLength
+				}, 1*time.Second, 10*time.Millisecond, "Pod is not added to PodsToDelete")
+			}
+		})
+	}
+}
+
+func Test_checkDeletedPod(t *testing.T) {
+	tests := []struct {
+		name           string
+		obj            interface{}
+		expectedResult *v1.Pod
+		expectedErr    string
+	}{
+		{
+			name:        "object is not DeletedFinalStateUnknown",
+			obj:         node,
+			expectedErr: "received unexpected object: ",
+		},
+		{
+			name:        "object in DeletedFinalStateUnknown is not Pod",
+			obj:         cache.DeletedFinalStateUnknown{Obj: node},
+			expectedErr: "DeletedFinalStateUnknown object is not of type Pod",
+		},
+		{
+			name:           "Valid case",
+			obj:            cache.DeletedFinalStateUnknown{Obj: pod1},
+			expectedResult: pod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := (&PodStore{}).checkDeletedPod(tt.obj)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expectedResult, got)
+			}
+		})
+	}
+}
+
+func Test_GetPodByIPAndTime(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	tests := []struct {
+		name           string
+		ip             string
+		startTime      time.Time
+		expectedResult *v1.Pod
+	}{
+		{
+			name:           "No Pod in the Pod store",
+			ip:             "1.3.5.7",
+			startTime:      refTime,
+			expectedResult: nil,
+		},
+		{
+			name:           "Find only one Pod in the Pod store -- correct startTime",
+			ip:             "5.6.7.8",
+			startTime:      refTime.Add(-time.Minute),
+			expectedResult: pod2,
+		},
+		{
+			name:           "Find only one Pod in the Pod store -- incorrect startTime ",
```

Comment:
there is an extra space at the end of the name

you use `--` as a separator here, but you use a period below. Maybe use a single `-` everywhere instead.

Reply:
Fixed. Thanks.